<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>YamlGenerator</class>
 <widget class="QWidget" name="YamlGenerator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1056</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_4">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>500</width>
        <height>875</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Loading a configuration file</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="2" column="1">
               <widget class="QLineEdit" name="lineEdit_filename"/>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>variable name:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>loadFile:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="lineEdit_variable0"/>
              </item>
              <item row="0" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QPushButton" name="pushButton">
                  <property name="text">
                   <string>url</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="horizontalSpacer">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QToolButton" name="toolButton_go_1">
                  <property name="text">
                   <string>G</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="toolButton_save_1">
                  <property name="text">
                   <string>S</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="plainTextEdit_templ_1">
              <property name="plainText">
               <string>YAML::Node config = YAML::LoadFile(&quot;%1&quot;);</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Basic Parsing and Node Editing</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_5">
            <item>
             <layout class="QGridLayout" name="gridLayout">
              <item row="1" column="1">
               <widget class="QLineEdit" name="lineEdit_variable"/>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="lineEdit_yamltext"/>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>variable name:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>yaml text:</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <item>
                 <spacer name="horizontalSpacer_5">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QToolButton" name="toolButton_go_2">
                  <property name="text">
                   <string>G</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QToolButton" name="toolButton_save_2">
                  <property name="text">
                   <string>S</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="plainTextEdit_templ_2">
              <property name="plainText">
               <string>YAML::Node config = YAML::LoadFile(&quot;%1&quot;);</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_4">
           <property name="title">
            <string>you can query the type of a node:</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_10">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <spacer name="horizontalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QToolButton" name="toolButton_go_3">
                <property name="text">
                 <string>G</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="toolButton_save_3">
                <property name="text">
                 <string>S</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="plainTextEdit_templ_3"/>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_6">
           <property name="title">
            <string>Building Nodes</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_6">
              <item>
               <spacer name="horizontalSpacer_4">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QToolButton" name="toolButton_go_4">
                <property name="text">
                 <string>G</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QToolButton" name="toolButton_save_4">
                <property name="text">
                 <string>S</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QPlainTextEdit" name="plainTextEdit_templ_4">
              <property name="plainText">
               <string>YAML::Node node;  // starts out as null
node[&quot;key&quot;] = &quot;value&quot;;  // it now is a map node
node[&quot;seq&quot;].push_back(&quot;first element&quot;);  // node[&quot;seq&quot;] automatically becomes a sequence
node[&quot;seq&quot;].push_back(&quot;second element&quot;);

node[&quot;mirror&quot;] = node[&quot;seq&quot;][0];  // this creates an alias
node[&quot;seq&quot;][0] = &quot;1st element&quot;;  // this also changes node[&quot;mirror&quot;]
node[&quot;mirror&quot;] = &quot;element #1&quot;;  // and this changes node[&quot;seq&quot;][0] - they're really the &quot;same&quot; node

node[&quot;self&quot;] = node;  // you can even create self-aliases
node[node[&quot;mirror&quot;]] = node[&quot;seq&quot;];  // and strange loops :)</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Converting To/From Native Data Types</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <spacer name="horizontalSpacer_6">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QToolButton" name="toolButton_go_5">
              <property name="text">
               <string>G</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QToolButton" name="toolButton_save_5">
              <property name="text">
               <string>S</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="plainTextEdit_templ_5"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QToolButton" name="toolButton_inter_clear">
           <property name="text">
            <string>*</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QCodeEditor" name="textEdit_inter"/>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCodeEditor</class>
   <extends>QTextEdit</extends>
   <header location="global">QCodeEditor</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>toolButton_go_1</tabstop>
  <tabstop>toolButton_save_1</tabstop>
  <tabstop>lineEdit_variable0</tabstop>
  <tabstop>lineEdit_filename</tabstop>
  <tabstop>plainTextEdit_templ_1</tabstop>
  <tabstop>toolButton_go_2</tabstop>
  <tabstop>toolButton_save_2</tabstop>
  <tabstop>lineEdit_variable</tabstop>
  <tabstop>lineEdit_yamltext</tabstop>
  <tabstop>plainTextEdit_templ_2</tabstop>
  <tabstop>toolButton_go_3</tabstop>
  <tabstop>toolButton_save_3</tabstop>
  <tabstop>plainTextEdit_templ_3</tabstop>
  <tabstop>toolButton_go_4</tabstop>
  <tabstop>toolButton_save_4</tabstop>
  <tabstop>plainTextEdit_templ_4</tabstop>
  <tabstop>toolButton_go_5</tabstop>
  <tabstop>toolButton_save_5</tabstop>
  <tabstop>plainTextEdit_templ_5</tabstop>
  <tabstop>toolButton_inter_clear</tabstop>
  <tabstop>textEdit_inter</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
